pkg/sql/sem/builtins/builtins.go

```diff
@@ -2506,6 +2510,40 @@ var jsonTypeOfImpl = tree.Builtin{
 	Info: "Returns the type of the outermost JSON value as a text string.",
 }
 
+var jsonBuildObjectImpl = tree.Builtin{
+	Types:        tree.VariadicType{VarType: types.Any},
+	ReturnType:   tree.FixedReturnType(types.JSON),
+	NullableArgs: true,
+	Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+		if len(args)%2 != 0 {
+			return nil, pgerror.NewError(pgerror.CodeInvalidParameterValueError,
+				"argument list must have even number of elements")
+		}
+
+		obj := make(map[string]interface{})
```

Comment:
You can use `json.Builder` to do this a little more efficiently thanks to @heyihong :) check #21015 for how it's used.